pkg/deploy/controller/deployment/controller.go

```diff
@@ -249,6 +250,10 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 		},
 	}
 
+	// MergeInfo will not overwrite values unless the flag OverwriteExistingDstKey is set.
+	util.MergeInto(pod.Labels, deploymentConfig.Template.Strategy.Labels, 0)
```

Comment:
How about something like `NeverOverwriteDstKey`? I personally don't mind if you leave it at `0` for now.